### DISCUSSION TITLE
Avoid using sudo in makeVarRun()

### DISCUSSION THREAD

**Discussion by IgorMilavec** [lima-vm/lima#2204] [Ideas]
### Description

Some security products obstruct the usage of sudo, causing lima to fail with something similar to:
`failed to run [sudo --user root --group wheel --non-interactive /bin/mkdir -m 775 -p /private/var/run/finch-lima]: stdout="", stderr="sudo: a password is required\n": exit status 1`.
This command is run from makeVarRun() even if the directory already exists with appropriate permissions.
Could lima run sudo mkdir only if os.stat() call fails? This way the user could pre-create the path.

**Comment by jandubois** [comment]
It could, but it would not really help, as it is just the first in a list of `sudo` commands Lima will execute to start the `socket_vmnet` daemon.

If you can't use `sudo`, then you will have to manage the daemon separately and not ask Lima to do it. See the https://github.com/lima-vm/socket_vmnet README about the ways to do this, e.g. to create a launchdaemon for the privileged `socket_vmnet` process.

**Reply by AkihiroSuda** [reply]
> If you can't use sudo, then you will have to manage the daemon separately and not ask Lima to do it. 

Or use `limactl create --vm-type vz --network vzNAT`.
This doesn't need sudo at all.

**Reply by jandubois** [reply]
Ah yes, I keep forgetting about that. However, it requires macOS 13.0.0 (Intel) or 13.3.0 (ARM) to use VZ.
